flow/display_list_benchmarks.cc

```diff
@@ -0,0 +1,1000 @@
+                 SkPath::Verb type) {
+  DisplayListBuilder builder;
+  size_t length = kFixedCanvasSize;
+  canvas_provider->InitializeSurface(length, length);
+  auto canvas = canvas_provider->GetSurface()->getCanvas();
+
+  SkPath path;
+
+  std::string label = VerbToString(type);
+  SkPoint center = SkPoint::Make(length / 2.0f, length / 2.0f);
+  float radius = length * 0.25f;
+  state.SetComplexityN(state.range(0));
+
+  MultiplyPath(path, type, center, 20, state.range(0), radius);
+
+  state.counters["VerbCount"] = path.countVerbs();
+
+  builder.drawPath(path);
+  auto display_list = builder.Build();
+
+  // We only want to time the actual rasterization.
+  for (auto _ : state) {
+    display_list->RenderTo(canvas);
+    canvas_provider->GetSurface()->flushAndSubmit(true);
+  }
+
+  auto filename = canvas_provider->BackendName() + "-DrawPath-" + label + "-" +
+                  std::to_string(state.range(0)) + ".png";
+  canvas_provider->Snapshot(filename);
+}
+
+// Returns a set of vertices that describe a circle that has a
+// radius of `radius` and total vertex count of approximately
+// `vertex_count`. The final number of vertices may differ as
+// we always return an even number of vertices from this method.
+//
+// The resulting vertices will describe a disc consisting of a series
+// of triangles with two vertices on the circumference of the disc,
+// and the final vertex being the center point of the disc.
+//
+// Each vertex colour will alternate through Red, Green, Blue and Cyan.
+sk_sp<SkVertices> GetTestVertices(SkPoint center,
+                                  float radius,
+                                  size_t vertex_count,
+                                  SkVertices::VertexMode mode,
+                                  size_t& final_vertex_count) {
+  size_t outer_vertex_count = vertex_count / 2;
+  std::vector<SkPoint> outer_points =
+      GetPolygonPoints(outer_vertex_count, center, radius);
+
+  std::vector<SkPoint> vertices;
+  std::vector<SkColor> colors;
+
+  switch (mode) {
+    case SkVertices::VertexMode::kTriangleFan_VertexMode:
+      vertices.push_back(center);
+      colors.push_back(SK_ColorCYAN);
+      for (size_t i = 0; i <= outer_points.size(); i++) {
+        vertices.push_back(outer_points[i % outer_points.size()]);
+        if (i % 3 == 0)
+          colors.push_back(SK_ColorRED);
+        else if (i % 3 == 1)
+          colors.push_back(SK_ColorGREEN);
+        else
+          colors.push_back(SK_ColorBLUE);
+      }
+      break;
+    case SkVertices::VertexMode::kTriangles_VertexMode:
+    case SkVertices::VertexMode::kTriangleStrip_VertexMode:
+      for (size_t i = 0; i <= outer_vertex_count; i++) {
+        vertices.push_back(outer_points[i % outer_points.size()]);
+        colors.push_back(i % 2 ? SK_ColorRED : SK_ColorGREEN);
+        vertices.push_back(center);
+        colors.push_back(i % 2 ? SK_ColorBLUE : SK_ColorCYAN);
```

Comment:
Oh yes, this is because the vertices were originally set up to do a doughnut shape and I just replaced the inner coordinates with the center point every time. I'll rework it.